chain-signatures/node/src/protocol/triple.rs

```diff
@@ -359,39 +343,35 @@ impl TripleManager {
         participants: &Participants,
         cfg: &ProtocolConfig,
     ) -> Result<Option<&mut TripleProtocol>, CryptographicError> {
-        if self.contains(id).await || self.gc.contains_key(&id) {
```

Comment:
The same logic here. We call `get_or_start_generation` after all messages are filtered against GC. That should be enough.